### PR TITLE
Add migration guide for microsoft-azure-authentication-msi-token-provider

### PR DESCRIPTION
**Pull request by pallavit**
# Description

This library is an old track1 beta library. All the scenarios supported in this library can be done using 'azure-identity', so I am adding a migration guide that helps users move to the new story instead.

We had a PR in the past to remove it -  https://github.com/Azure/azure-sdk-for-java/pull/9001 but abandoned it.
We had a customer hit into issue while using it - https://github.com/Azure/azure-sdk-for-java/issues/25290

# All SDK Contribution checklist:
- [X] **The pull request does not introduce [breaking changes]**
- [] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [X] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#general-guidelines)
- [X] Title of the pull request is clear and informative.
- [] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](../../CONTRIBUTING.md#building-and-unit-testing)
- [] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/authorization/microsoft-azure-authentication-msi-token-provider/migration-guide.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+<!-- Links -->
+[Guidelines]: https://azure.github.io/azure-sdk/general_introduction.html
+[GuidelinesJava]: https://azure.github.io/azure-sdk/java_introduction.html
+[GuidelinesJavaDesign]: https://azure.github.io/azure-sdk/java_introduction.html#namespaces
+[project-reactor]: https://projectreactor.io/
+[README-Samples]: https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/identity/azure-identity/src/samples/README.md
+[README]: https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/identity/azure-identity/README.md
+
+![Impressions](https://azure-sdk-impressions.azurewebsites.net/api/impressions/azure-sdk-for-java%2Fsdk%2Feventhubs%2Fazure-messaging-eventhubs%2Fmigration-guide.png)
```
Comment (srnagar) [inline sdk/authorization/microsoft-azure-authentication-msi-token-provider/migration-guide.md:99]: Fix the path of the impressions tag.

File: sdk/authorization/microsoft-azure-authentication-msi-token-provider/README.md [hunk 2]
```diff
@@ -1,70 +0,0 @@
-# What is this?
```
Comment (srnagar) [inline sdk/authorization/microsoft-azure-authentication-msi-token-provider/README.md:1]: We should keep the README to just add a deprecation disclaimer as we have in this [Event Grid](https://github.com/Azure-Samples/event-grid-java-publish-consume-events/blob/master/README.md) README and include a link to the migration guide. Another advantage of README is that it also shows the markdown on the landing page, so, it's immediately visible.

File: sdk/authorization/pom.xml [hunk 3]
```diff
@@ -8,7 +8,4 @@
   <artifactId>azure-authorization-service</artifactId>
```
Comment (srnagar) [inline sdk/authorization/pom.xml:8]: Do we want to keep this pom.xml and ci.yml in this sdk/authorization directory? There are no other artifacts here.

File: sdk/authorization/microsoft-azure-authentication-msi-token-provider/migration-guide.md [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+
+    // Azure SDK client builders accept the credential as a parameter
+    SecretClient client = new SecretClientBuilder()
+        .vaultUrl("https://{YOUR_VAULT_NAME}.vault.azure.net")
+        .credential(managedIdentityCredential)
+        .buildClient();
+}
+```
+
+#### Authentication with user assigned managed identity
```
Comment (g2vinay) [inline sdk/authorization/microsoft-azure-authentication-msi-token-provider/migration-guide.md:69]: This title doesn't match the sample below, either swap the title or sample with above.

File: sdk/authorization/microsoft-azure-authentication-msi-token-provider/migration-guide.md [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+
+### Group id, artifact id, and package names
+
+Group ids, artifact ids, and package names for the modern Azure client libraries for Java have changed. They follow the [Java SDK naming guidelines][GuidelinesJavaDesign]. Each will have the group id `com.azure`, an artifact id following the pattern `azure-[area]-[service]`, and the root package name `com.azure.[area].[Service]`. The legacy clients have a group id `com.microsoft.azure` and their package names followed the pattern `com.microsoft. Azure.[service]`. This provides a quick and accessible means to help understand, at a glance, whether you are using modern or legacy clients.
+
+In azure-identity, the modern client libraries have packages and namespaces that begin with `com.azure.identity`. The legacy library has package names starting with `com.microsoft.azure`.
+
+### Authentication with managed identity
+`microsoft-azure-authentication-msi-token-provider` allows authenticating using system and user assigned managed identity. It supports Azure App Services and Azure Virtual Machines.
+You can use `azure-identity` to do the same in the following way.
```
Comment (g2vinay) [inline sdk/authorization/microsoft-azure-authentication-msi-token-provider/migration-guide.md:47]: link to the azure-identity README from here.

**Comment by pallavit** [comment]
/cc: @srnagar, @g2vinay thank you for the review, I have incorporated the feedback.

File: sdk/authorization/microsoft-azure-authentication-msi-token-provider/readme.md [hunk 6]
```diff
@@ -1,3 +1,6 @@
+***DISCLAIMER: The samples in this readme are for azure-authentication-msi-token-provider. This SDK has been deprecated.
+Please use [azure-identity](https://github.com/Azure/azure-sdk-for-java/tree/main/sdk/identity/azure-identity/readme.md) for authentication support. If you have existing code that uses the library, please use the [Migration Guide](https://github.com/Azure/azure-sdk-for-java/tree/main/sdk/authorization/microsoft-azure-authentication-msi-token-provider/migration-guide.md) instead.***
```
Comment (srnagar) [inline sdk/authorization/microsoft-azure-authentication-msi-token-provider/readme.md:2]:
```suggestion
Please use [azure-identity](https://github.com/Azure/azure-sdk-for-java/tree/main/sdk/identity/azure-identity/readme.md) for authentication support. If you have existing code that uses the library, please use the [Migration Guide](https://github.com/Azure/azure-sdk-for-java/tree/main/sdk/authorization/microsoft-azure-authentication-msi-token-provider/migration-guide.md) to migrate to `azure-identity`.***
```

**Review by srnagar — APPROVED** [review]

**Pull request merged**